mlflow/models/evaluation/utils/trace.py

```diff
@@ -24,87 +35,118 @@ def configure_autologging_for_evaluation(enable_tracing: bool = True):
-                    # when user explicitly enables it for a specific flavor via
-                    # mlflow.<flavor>.autolog(), the global disablement does not take precedence,
-                    # which is intended.
-                    # However, in Databricks, sometimes mlflow.<flavor>.autolog() is automatically
-                    # called in the kernel startup, which is confused with the user's action. In
-                    # such cases, even when user disables autologging globally, the flavor-specific
-                    # autologging remains enabled. We are going to fix the Databricks side issue,
-                    # but until then we need to check the global configuration explicitly here.
-                    trace_disabled = global_config_snapshot.get("mlflow", {}).get("disable", False)
-
-                    new_config |= {"log_traces": not trace_disabled, "silent": True}
+                    new_config |= {"log_traces": True, "silent": True}
                     _kwargs_safe_invoke(autolog, new_config)
-                    trace_enabled_flavors.append(flavor)
-                elif flavor in AUTOLOGGING_INTEGRATIONS:
-                    # For flavors that do not support tracing, disable autologging
-                    autolog(disable=True)
-
-                flavor_to_original_config[flavor] = original_config
 
-        except Exception as e:
-            if isinstance(e, ImportError):
-                _logger.debug(
-                    f"Flavor {flavor} is not installed. Skip updating autologging. Error: {e!r}"
-                )
-            else:
-                _logger.info(
-                    f"Failed to update autologging configuration for flavor {flavor}. Error: {e!r}"
-                )
+                    global _IS_TRACE_MESSAGE_DISPLAYED
+                    if not _IS_TRACE_MESSAGE_DISPLAYED:
+                        _logger.info(
+                            "Tracing is temporarily enabled during the model evaluation "
+                            "for computing some metrics and debugging. To disable tracing, call "
+                            "`mlflow.autolog(disable=True)`."
+                        )
+                        _IS_TRACE_MESSAGE_DISPLAYED = True
+                else:
+                    autolog(disable=True)
 
-            # Autologging configuration might be updated before the exception is raised,
-            # which needs to be reverted.
-            AUTOLOGGING_INTEGRATIONS.pop(flavor, None)
+            except Exception:
+                _logger.debug(f"Failed to update autologging config for {flavor}.", exc_info=True)
 
-    if trace_enabled_flavors:
-        _logger.info(
-            "Tracing is temporarily enabled during the model evaluation for computing some "
-            "metrics and debugging. To disable tracing, call `mlflow.autolog(disable=True)`."
-        )
+        try:
+            module = FLAVOR_TO_MODULE_NAME[flavor]
```

Comment:
can this line fail? If not, can we pull it out from try-catch?

Reply:
No, you're right, we can pull this off and only wrap lines related to import hooks.